app/controllers/vaccines_controller.rb

```diff
@@ -32,104 +34,107 @@ def index
+        render(json: { error: error_message }, status: :bad_request) && return
+      end
     end
+  end
 
-    # Check if user has access to patient
-    unless current_user.get_patient(patient_id)
-      error_message = "User does not have access to Patient with ID: #{patient_id}"
-      render(json: { error: error_message }, status: :forbidden) && return
+  def destroy
+    ActiveRecord::Base.transaction do
+      if @vaccine.destroy
+        reason = params.permit(:delete_reason)[:delete_reason]
+        comment = "User deleted a vaccine (ID: #{@vaccine.id}"
+        comment += ", Vaccine Group: #{@vaccine.group_name}" unless @vaccine.group_name.blank?
+        comment += ", Product Name: #{@vaccine.product_name}" unless @vaccine.product_name.blank?
+        comment += ", Administration Date: #{@vaccine.administration_date.strftime('%m/%d/%Y')}" unless @vaccine.administration_date.blank?
+        comment += ", Dose Number: #{@vaccine.dose_number}" unless @vaccine.dose_number.blank?
+        comment += "). Reason: #{reason}."
+        History.vaccination_edit(patient: @patient.id,
+                                 created_by: current_user.email,
+                                 comment: comment)
+      else
+        # Handle case where vaccine update failed
+        error_message = 'Vaccination was unable to be deleted.'
+        render(json: { error: error_message }, status: :bad_request) && return
+      end
     end
+  end
 
-    # Get vaccine to update
-    vaccine = Vaccine.find_by(id: vaccine_id)
+  private
 
-    # Handle case where vaccine cannot be found
-    render(json: { error: "Vaccination with ID #{vaccine_id} cannot be found." }, status: :bad_request) && return unless vaccine
+  def check_can_create
+    return head :forbidden unless current_user.can_create_patient_vaccines?
+  end
 
-    # Update the vaccine record
-    update_params = {
-      group_name: group_name,
-      product_name: product_name,
-      administration_date: administration_date,
-      dose_number: dose_number,
-      notes: notes,
-      patient_id: patient_id
-    }
+  def check_can_edit
+    return head :forbidden unless current_user.can_edit_patient_vaccines?
+  end
 
-    # Handle vaccine update success or failure
-    if vaccine.update(update_params)
-      # Create history item on successful update
-      History.vaccination_edit(
-        patient: params.permit(:patient_id)[:patient_id],
-        created_by: current_user.email,
-        comment: "User edited a vaccination (ID: #{vaccine_id})."
-      )
-    else
-      # Handle case where vaccine update failed
-      error_message = 'Vaccination was unable to be updated. '
-      error_message += "Errors: #{format_model_validation_errors(vaccine).join(', ')}" if vaccine&.errors
-      render(json: { error: error_message }, status: :bad_request) && return
+  def check_patient
+    # Check if Patient ID is valid
+    patient_id = params.permit(:patient_id)[:patient_id]&.to_i
```

Comment:
let's require `patient_id` instead of permitting it

Reply:
good call, fixed it here and in the labs controller here: https://github.com/SaraAlert/SaraAlert/pull/897/commits/3508fcb56bf795c52006ece55807f860b497832d